src/MICore/Debugger.cs

```diff
@@ -578,20 +578,22 @@ public Task CmdBreak(BreakRequest request)
             return CmdBreakInternal();
         }
 
-        internal bool IsLocalGdb()
+        protected bool IsLocalLaunchUsingServer()
         {
-            return (this.MICommandFactory.Mode == MIMode.Gdb &&
-               this._launchOptions is LocalLaunchOptions &&
-               String.IsNullOrEmpty(((LocalLaunchOptions)this._launchOptions).MIDebuggerServerAddress));
+            return (_launchOptions is LocalLaunchOptions &&
+                (!String.IsNullOrWhiteSpace(((LocalLaunchOptions)_launchOptions).MIDebuggerServerAddress) ||
+                 !String.IsNullOrWhiteSpace(((LocalLaunchOptions)_launchOptions).DebugServer)));
```

Comment:
Do we want to do a quick/dirty string compare for `"setupCommands"` also to see if any start with `-target-select remote` or `-target-select extended-remote` and add that as an option here?

Also, do we want to do this one validation one time instead of any time that this method is called?

Reply:
Actually, there is a way to know if someone did a `remote` or `extended-remote`. You will get a response from gdb `^connected`. Scanning in `setupCommands` is not a reliable method, I thought. They could have done that using 'mi' commands or even non-mi commands. Looking for `^connected` was my first implementation and it worked but we would know about that after the session starts. My recent implementation using launch properties, we will know even before the session starts.

Yes, I suppose, this could be validated once. Did not want to change too much from what the existing code did. I just refactored it and added the additional check. Do you want me to do that? This method is used so infrequently so....

Reply:
>Yes, I suppose, this could be validated once. Did not want to change too much from what the existing code did. I just refactored it and added the additional check. Do you want me to do that? This method is used so infrequently so....

The value won't change in subsequent checks so my gut is to do it one time and cache it but we can see if this way is performant enough and make a change in the future if necessary.

Reply:
Yup, the same thing can be said about the old `IsLocalGdb()` and `IsRemoteGdb()`. This code is only executed during startup and when the user hits pause/terminate. Perhaps keeping less state is also a good thing?

Reply:
True. Like you mentioned in the testing document, theres a lot of iffy state management that we try and do such as the stopAtEntry work that isn't 100% foolproof.